workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncherWindow.cs

```diff
@@ -0,0 +1,598 @@
+                        {
+                            UpdateSimulatedDeploymentNames(copy);
+                        }
+
+                        GUILayout.Space(10);
+
+                        EditorGUILayout.LabelField("Simulated Player Deployments");
+
+                        for (var i = 0; i < copy.SimulatedPlayerDeploymentConfig.Count; i++)
+                        {
+                            var simConfig = copy.SimulatedPlayerDeploymentConfig[i];
+                            var (shouldRemove, updated) = DrawSimulatedConfig(i, simConfig);
+
+                            GUILayout.Space(5);
+
+                            if (shouldRemove)
+                            {
+                                copy.SimulatedPlayerDeploymentConfig.RemoveAt(i);
+                                i--;
+                                UpdateSimulatedDeploymentNames(copy);
+                            }
+                            else
+                            {
+                                copy.SimulatedPlayerDeploymentConfig[i] = updated;
+                            }
+                        }
+                    }
+                }
+
+                using (new EditorGUILayout.HorizontalScope())
+                {
+                    if (foldoutState)
+                    {
+                        GUILayout.FlexibleSpace();
+
+                        if (GUILayout.Button("Add simulated player deployment"))
+                        {
+                            var newSimPlayerDepl = new SimulatedPlayerDeploymentConfig();
+                            newSimPlayerDepl.TargetDeploymentName = config.Deployment.Name;
+                            newSimPlayerDepl.Name = $"{config.Deployment.Name}_sim{config.SimulatedPlayerDeploymentConfig.Count + 1}";
+
+                            copy.SimulatedPlayerDeploymentConfig.Add(newSimPlayerDepl);
+                        }
+                    }
+                }
+
+                if (errors.Any())
+                {
+                    EditorGUILayout.HelpBox($"This deployment configuration has the following errors:\n\n{errors.FormatErrors()}", MessageType.Error);
+                }
+
+                if (check.changed)
+                {
+                    stateManager.SetStateObject(copy.Deployment.Name.GetHashCode(), foldoutState);
+                }
+            }
+
+            DrawHorizontalLine(5);
+
+            return (false, copy);
+        }
+
+        private void RenderBaseDeploymentConfig(BaseDeploymentConfig source, BaseDeploymentConfig dest)
+        {
+            using (new EditorGUI.DisabledScope(source is SimulatedPlayerDeploymentConfig))
+            {
+                dest.Name = EditorGUILayout.TextField("Deployment Name", source.Name);
+            }
+
+            dest.SnapshotPath = EditorGUILayout.TextField("Snapshot Path", source.SnapshotPath);
+            dest.LaunchJson = EditorGUILayout.TextField("Launch Config", source.LaunchJson);
```

Comment:
can we add a check that it needs to be a json file? currently I can try to use a snapshot as a launch config :trollface:

Reply:
That's on you